src/pal/src/include/pal/utils.h

```diff
@@ -129,5 +129,29 @@ void UTIL_SetLastErrorFromMach(kern_return_t MachReturn);
 #ifdef __cplusplus
 }
 #endif // __cplusplus
-
+class StringHolder
+   {
+       private:
+           LPSTR data;
+       public:
+        StringHolder() : data(NULL) { }
+        ~StringHolder() 
+        { if(data)  
```

Comment:
You don't need to check the data for not being NULL, free handles NULL just fine.